apps/worker/src/app/workflow/usecases/send-message/send-message-voice.usecase.ts

```diff
@@ -0,0 +1,338 @@
+import { Injectable } from '@nestjs/common';
+import * as Sentry from '@sentry/node';
+import { ModuleRef } from '@nestjs/core';
+
+import {
+  MessageRepository,
+  NotificationStepEntity,
+  SubscriberRepository,
+  MessageEntity,
+  IntegrationEntity,
+} from '@novu/dal';
+import { ChannelTypeEnum, LogCodeEnum, ExecutionDetailsSourceEnum, ExecutionDetailsStatusEnum } from '@novu/shared';
+import {
+  InstrumentUsecase,
+  DetailEnum,
+  SelectIntegration,
+  CompileTemplate,
+  CompileTemplateCommand,
+  VoiceFactory,
+  GetNovuProviderCredentials,
+  SelectVariant,
+  ExecutionLogRoute,
+  ExecutionLogRouteCommand,
+} from '@novu/application-generic';
+
+import { CreateLog } from '../../../shared/logs';
+import { SendMessageCommand } from './send-message.command';
+import { SendMessageBase } from './send-message.base';
+import { PlatformException } from '../../../shared/utils';
+
+@Injectable()
+export class SendMessageVoice extends SendMessageBase {
+  channelType = ChannelTypeEnum.VOICE;
+
+  constructor(
+    protected subscriberRepository: SubscriberRepository,
+    protected messageRepository: MessageRepository,
+    protected createLogUsecase: CreateLog,
+    protected executionLogRoute: ExecutionLogRoute,
+    private compileTemplate: CompileTemplate,
+    protected selectIntegration: SelectIntegration,
+    protected getNovuProviderCredentials: GetNovuProviderCredentials,
+    protected selectVariant: SelectVariant,
+    protected moduleRef: ModuleRef
+  ) {
+    super(
+      messageRepository,
+      createLogUsecase,
+      executionLogRoute,
+      subscriberRepository,
+      selectIntegration,
+      getNovuProviderCredentials,
+      selectVariant,
+      moduleRef
+    );
+  }
+
+  @InstrumentUsecase()
+  public async execute(command: SendMessageCommand) {
+    const overrideSelectedIntegration = command.overrides?.voice?.integrationIdentifier;
+
+    const integration = await this.getIntegration({
+      organizationId: command.organizationId,
+      environmentId: command.environmentId,
+      channelType: ChannelTypeEnum.VOICE,
+      userId: command.userId,
+      identifier: overrideSelectedIntegration as string,
+      filterData: {
+        tenant: command.job.tenant,
+      },
+    });
+
+    Sentry.addBreadcrumb({
+      message: 'Sending VOICE',
+    });
+
+    const step: NotificationStepEntity = command.step;
+
+    if (!step.template) throw new PlatformException(`Unexpected error: VOICE template is missing`);
+
+    const { subscriber } = command.compileContext;
+    const template = await this.processVariants(command);
+    await this.initiateTranslations(command.environmentId, command.organizationId, subscriber.locale);
+
```

Comment:
i'm not sure what exactly this does as its not open sourced, not sure if this is relavant for voice.